sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Metrics/ExecutableLinesWalkerTest.cs

```diff
@@ -438,5 +434,156 @@ int FooProperty
                     }
                 }");
         }
+
+        [TestMethod]
+        public void Test_23_Constructor_ExcludeFromCodeCoverage()
+        {
+            AssertLinesOfCode(
+              @"using System.Diagnostics.CodeAnalysis;
+                class Program
+                {
+                    int count;
+
+                    [ExcludeFromCodeCoverage]
+                    public Program() : this(1)
+                    {
+                        count = 123;            // excluded
+                    }
+
+                    public Program(int initialCount)
+                    {
+                        count = initialCount;   // +1
+                    }
+                }",
+                14);
+        }
+
+        [TestMethod]
+        public void Test_24_Property_ExcludeFromCodeCoverage()
+        {
+            AssertLinesOfCode(
+              @"using System.Diagnostics.CodeAnalysis;
+                class EventClass
+                {
+                    private int _value;
+
+                    [System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
+                    public int Value1
+                    {
+                        get { return _value; }      // Excluded
+                        set { _value = value; }     // Excluded
+                    }
+
+                    public int Value2
+                    {
+                        get { return _value; }      // +1
+                        [System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
+                        set { _value = value; }     // Excluded
+                    }
+
+                    public int Value3
+                    {
+                        get { return _value; }      // +1
+                        set { _value = value; }     // +1
+                    }
+                }",
+                15, 22, 23);
+        }
+
+
+        [TestMethod]
+        public void Test_25_Event_ExcludeFromCodeCoverage()
+        {
+            AssertLinesOfCode(
+              @"using System.Diagnostics.CodeAnalysis;
+                class EventClass
+                {
+                    private System.EventHandler _explicitEvent;
+
+                    [System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
+                    public event System.EventHandler ExplicitEvent1
+                    {
+                        add { _explicitEvent += value; }        // Excluded
+                        remove { _explicitEvent -= value; }     // Excluded
+                    }
+
+                    public event System.EventHandler ExplicitEvent2
+                    {
+                        add { _explicitEvent += value; }        // +1 ?? TODO check
```

Comment:
Presumably we know what the expected behaviour is now so this TODO can be removed?